### PR TITLE
Use createMock when available

### PR DESCRIPTION
**Pull request by greg0ire**
I am targeting this branch, because this is pedantic (related to the test suite).

<!--
    Specify which issues will be fixed/closed.

### CODE REVIEW

**Comment by OskarStark** [comment]
#814 is merged now, please rebase and fix the conflicts

**Comment by greg0ire** [comment]
@OskarStark do you know why the build might be failing? Since you wrote similar PRs...

**Comment by greg0ire** [comment]
Oh it must be related to disabling the constructors

File: Tests/Block/PageListBlockServiceTest.php [hunk 1]
```diff
@@ -29,7 +29,7 @@ protected function setUp()
     {
         parent::setUp();
 
-        $this->pageManager = $this->getMock('Sonata\PageBundle\Model\PageManagerInterface');
+        $this->pageManager = $this->createMock('Sonata\PageBundle\Model\PageManagerInterface');
```
Comment (OskarStark) [inline Tests/Block/PageListBlockServiceTest.php]: does AbstractBlockServiceTestCase extend the new TestCase? I mean, related to the failing travis build (--prefer-lowest)?

Comment (greg0ire) [inline Tests/Block/PageListBlockServiceTest.php]: It doesn't on old versions indeed, let me find out what to use as a new requirement

Comment (greg0ire) [inline Tests/Block/PageListBlockServiceTest.php]: Oh wait, it does't have it on new versions either, it just has its own `createMock`

Comment (OskarStark) [inline Tests/Block/PageListBlockServiceTest.php]: Looks good, but StyleCI is stuck

**Pull request merged**